### PR TITLE
Deploy Automation

### PR DESCRIPTION
**Pull request by pjstein**
Automatically deploy the backend to dev when there is a push to dev. Automatically deploy the backend to production when there is a push to master.

### CODE REVIEW

File: infra/resources.tf [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      "arn:aws:s3:::${var.deploy_bucket}",
+      "arn:aws:s3:::${var.deploy_bucket}/*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:ListAllMyBuckets",
+      "s3:CreateBucket"
```
Comment (curiousjazz77) [inline infra/resources.tf:59]: LGTM. Do we also need put permission here?

**Review by curiousjazz77 — APPROVED** [review]

**Comment by curiousjazz77** [comment]
I heard that Airtable has a "backup" feature that could be rolled into a deploy process. In case we want backups that we can restore from down the road.

File: infra/resources.tf (hunk 1 above)
Comment (pjstein) [reply infra/resources.tf:59]: Which permission?

**Comment by pjstein** [comment]
@curiousjazz77 where'd you see information about the airtable backups? I'm not finding things on Google.

File: infra/resources.tf (hunk 1 above)
Comment (curiousjazz77) [reply infra/resources.tf:59]: "s3:PutObject"

I'm unsure of the moving pieces here though.

**Comment by curiousjazz77** [comment]
> @curiousjazz77 where'd you see information about the airtable backups? I'm not finding things on Google.

There seem to be a few options, but al of them are rather manual. Looks like you can save the database as JSON (from the API) or csv. Airtable itself doesn't help with a restoration process:

https://community.airtable.com/t/airtable-backup-recovery-strategies-simply-a-nightmare/25777/8

**Comment by curiousjazz77** [comment]
This option looks more viable and is integrated into AWS: https://community.airtable.com/t/unly-releases-an-open-source-software-for-airtable-managed-backups/26752

demo starts at 22:50
https://www.youtube.com/watch?v=DR7zgsoJkTg&feature=youtu.be&t=1371

**Pull request merged**